hudi-common/src/main/java/org/apache/hudi/common/table/log/HoodieMergedLogRecordScanner.java

```diff
@@ -153,9 +154,29 @@ protected void processNextRecord(HoodieRecord<? extends HoodieRecordPayload> hoo
   }
 
   @Override
-  protected void processNextDeletedKey(HoodieKey hoodieKey) {
-    records.put(hoodieKey.getRecordKey(), SpillableMapUtils.generateEmptyPayload(hoodieKey.getRecordKey(),
-        hoodieKey.getPartitionPath(), getPayloadClassFQN()));
+  protected void processNextDeletedKey(DeleteKey deleteKey) {
+    String key = deleteKey.getRecordKey();
+    if (records.containsKey(key)) {
```

Comment:
Why making 2 hash-table lookups, instead of 1?

Reply:
Sorry, can't get your point.

Reply:
`containsKey` and then `get` are the 2 lookups i was referring to. Why not instead do just 1 lookup?
```
value = records.get(key);
if (value != null) { ... }
```

Reply:
yes, reduce 1 lookup, line 140 is the same

Reply:
+1 . Don't want to harp on this though.